pylj/forcefields.py

```diff
@@ -140,20 +138,28 @@ def square_well(dr, constants, force=False):
 
     Returns
     -------
-    float:
-        The potential energy or force between the particles.
+    float: array_like
+        The potential energy between the particles.
     '''
+    if not isinstance(dr, np.ndarray):
+        if isinstance(dr, list):
+            dr = np.array(dr, dtype='float')
+        elif isinstance(dr, float):
+            dr = np.array([dr], dtype='float')
+
     if force:
-        if dr < constants[1]:
-            return None
-        elif constants[1] <= dr and dr <= constants[2] * constants[1]:
-            return 0
-        elif dr >= constants[2] * constants[1]:
+        if all(constants[1] <= dr) and all(dr <= constants[2] * constants[1]):
+            raise ValueError(
+                "Force is infinite at sigma <= dr < lambda * sigma")
+        else:
             return 0
     else:
-        if dr < constants[1]:
-            return float('inf')
-        elif constants[1] <= dr and dr <= constants[2] * constants[1]:
-            return -constants[0]
-        elif dr >= constants[2] * constants[1]:
-            return 0
+        E = []
+        for r in dr:
```

Comment:
I suspect that this may be achieved with np intrinsic function something like 

```
e[np.where(dr < constants[1])] = np.inf
```
etc. 

(I am not 100% on that code I am doing it from memory in a pub in sweden but I think that is the jist)